skpro/regression/base/_base.py

```diff
@@ -72,7 +72,7 @@ def __rmul__(self, other):
         else:
             return NotImplemented
 
-    def fit(self, X, y, C=None):
+    def fit(self, X, y, C=None, sample_weight=None):
```

Comment:
as said, good idea, but should be in a separate PR. Could you also open an issue to track adding sample weights?

What we should do (please copy this recommendation in an issue):

* add a tag `capability:sample_weight` that ells the user whether the weights are used non-trivially
* add tests in `TestAllRegressors` that actually passes sample weights, to see that nothing breaks. In particular, nothing should break for any estimator when passed, the current boilerplate would lead to most estimators breaking, as the argument is passed on to `_fit`!
* in particular, the boilerplate should check the tag, and pass `sample_weight` on to `_fit` only if the tag has value `True`.
* in the tests, we should check that `_fit` does have the arg `sample_weight` if the tag is set `True`.